### PR TITLE
Set SEMGREP_USER_AGENT_APPEND to distinguish environment

### PR DESCRIPTION
**Pull request by mschwager**
For https://github.com/returntocorp/enterprise/issues/359.

Is there a good way to test that this is working?

### CODE REVIEW

File: src/semgrep_agent/semgrep.py [hunk 1]
```diff
@@ -37,7 +37,8 @@
 if TYPE_CHECKING:
     from semgrep_agent.semgrep_app import Scan
 
-semgrep = sh.semgrep.bake(_ok_code={0, 1, 2}, _tty_out=False)
+ua_environ = {**os.environ, **{'SEMGREP_USER_AGENT_APPEND', '(Agent)'}}
```
Comment (underyx) [inline src/semgrep_agent/semgrep.py]:
```suggestion
ua_environ = {**os.environ, **{'SEMGREP_USER_AGENT_APPEND': '(Agent)'}}
```
Did you mean this?

Comment (mschwager) [reply src/semgrep_agent/semgrep.py]: Yes, good catch!

Comment (underyx) [reply src/semgrep_agent/semgrep.py]:
```suggestion
ua_environ = {"SEMGREP_USER_AGENT_APPEND": "(Agent)", **os.environ}
```

This should be nicer though anyway. Also allows overriding with an env var visible to semgrep-agent if you really want to.

File: src/semgrep_agent/semgrep.py [hunk 2]
```diff
@@ -37,7 +37,8 @@
 if TYPE_CHECKING:
     from semgrep_agent.semgrep_app import Scan
 
-semgrep = sh.semgrep.bake(_ok_code={0, 1, 2}, _tty_out=False)
+ua_environ = {**os.environ, **{'SEMGREP_USER_AGENT_APPEND': '(Agent)'}}
```
Comment (underyx) [inline src/semgrep_agent/semgrep.py]:
```suggestion
ua_environ = {"SEMGREP_USER_AGENT_APPEND": "(Agent)", **os.environ}
```

This should be nicer though anyway. Also allows overriding with an env var visible to semgrep-agent if you really want to.

Comment (underyx) [reply src/semgrep_agent/semgrep.py]: Sorry, I didn't write this comment in time previously 😅

**Comment by underyx** [comment]
As for testing, I'm not sure — perhaps it's enough to see if `(Agent)` user agents start showing up in our server logs and analytics?

**Comment by mschwager** [comment]
@DrewDennison mentioned we could check live analytics via Segment. Would the CI test run with this new functionality show up there for us to confirm?

**Comment by mschwager** [comment]
Confirmed in Segment: `"Semgrep/0.25.0 (Agent)"` we should be good to go.

**Review by underyx — APPROVED** [review]

**Pull request merged**